src/drive/web/modules/actions/index.jsx

```diff
@@ -0,0 +1,200 @@
+/* global __TARGET__ */
+import React, { useContext } from 'react'
+import { useDispatch } from 'react-redux'
+import { models, useClient } from 'cozy-client'
+import { SharingContext, ShareModal } from 'cozy-sharing'
+import { getTracker } from 'cozy-ui/transpiled/react/helpers/tracker'
+import { ModalContext } from 'drive/lib/ModalContext'
+import { useRouter } from 'drive/lib/RouterContext'
+
+import DeleteConfirm from 'drive/web/modules/drive/DeleteConfirm'
+import MoveModal from 'drive/web/modules/move/MoveModal'
+import { EditDocumentQualification } from 'cozy-scanner'
+import ShareMenuItem from 'drive/web/modules/drive/ShareMenuItem'
+import MakeAvailableOfflineMenuItem from 'drive/web/modules/drive/MakeAvailableOfflineMenuItem'
+import DestroyConfirm from 'drive/web/modules/trash/components/DestroyConfirm'
+
+import { isIOSApp } from 'cozy-device-helper'
+import { startRenamingAsync } from 'drive/web/modules/drive/rename'
+
+const { file: fileModel } = models
+const { isFile } = fileModel
+
+import {
+  isAnyFileReferencedByAlbum,
+  exportFilesNative,
+  downloadFiles,
+  trashFiles,
+  openFileWith,
+  restoreFiles
+} from './utils'
+
+export const share = ({ hasWriteAccess }) => {
+  const { pushModal, popModal } = useContext(ModalContext)
```

Comment:
I think we are going in the right direction with the modular actions.

But, I am unconfortable with the fact that we use hooks outside rendering functions. Hooks are really weird and I do not think that it's a good idea to have them far away from the rendering function. I think an object containing { modalContext, client } could be passed to actions.

Reply:
I went with that approach, the options object is created in the component that calls `useAction`, so all the hook calls are close to the actions. And views that only use a few options can import just what they need in the options.